.github/workflows/compile_protos.yml

```diff
@@ -5,111 +5,89 @@ concurrency:
   cancel-in-progress: true
 
 on:
-  workflow_run:
-    workflows: ["On PR Approved"]
-    types:
-      - completed
+  pull_request_target:
+    types: ['synchronize', 'labeled']
+    branches:
+    - main
+
+env:
+  # guards against running this action on its own previous commit
+  is-auto-commit:
+  commit-message:
+  commit-author:
+  GRPC_WEB_VERSION: '1.5.0'
 
 jobs:
+  unlabel:
+    # This unlabel job runs on manual commits only.
+    # (because commits generated with GITHUB_TOKEN don't rerun workflow per https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow)
+    # note: this job doesn't need a 'safe to test' label because it doesn't check out code
+    runs-on: ubuntu-latest
+    steps:
+    - name: remove label
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GH_REPO: ${{ github.repository }}
+      run: gh pr edit ${{ github.event.pull_request.number }} --remove-label protos-compiled
+
   compile-protos:
-    if: >
-      github.event.workflow_run.event == 'pull_request_review' &&
-      github.event.workflow_run.conclusion == 'success'
-    runs-on: [self-hosted, x64]
-    container:
-      image: ghcr.io/viamrobotics/canon:amd64-cache
-      options: --platform linux/amd64
+    # run after unlabel so that ordering of remove-label, add-label is consistent
+    needs: unlabel
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') && contains(github.event.pull_request.labels.*.name, 'ready-for-protos')
+    runs-on: ubuntu-latest
     env:
       CI_COMMIT_MESSAGE_PREFIX: Built new protos from
       CI_COMMIT_AUTHOR: github-actions
-      LABEL_NAME: protos-compiled
     steps:
-      # Download PR info from approval workflow and checkout branch
-      - name: Download artifact
-        uses: dawidd6/action-download-artifact@v2
-        with:
-          run_id: ${{github.event.workflow_run.id }}
-      - name: Get PR Repo
-        id: get-pr-repo
-        uses: actions/github-script@v6
+      - uses: actions/checkout@v4
         with:
-          result-encoding: string
-          script: |
-            var fs = require('fs');
-            var repo = String(fs.readFileSync('./pr/repo')).trim();
-            return repo
-      - name: Get PR Ref
-        id: get-pr-ref
-        uses: actions/github-script@v6
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 2
```

Comment:
Why `fetch-depth: 2`?

Reply:
oops good catch. I added in cefbcd59, presumably because my brain saw `git log -1`. removed in f6b6fa1e

Reply:
and reran workflow [here](https://github.com/viamrobotics/api/actions/runs/8714800192/job/23905631080?pr=481), :heavy_check_mark: passed